### PR TITLE
Fixed-size data types use

### PR DESCRIPTION
**Pull request by fckt**
use of H256 and H160 instead of plain u8-arrays

### CODE REVIEW

File: src/ext.rs [hunk 1]
```diff
@@ -81,7 +84,7 @@ pub fn create(endowment: &[u8; 32], code: &[u8]) -> Result<[u8; 20], Error> {
     }
 }
 
-pub fn call(address: &[u8; 20], value: &[u8; 32], input: &[u8], result: &mut [u8]) -> Result<(), Error> {
+pub fn call(address: &Address, value: &[u8; 32], input: &[u8], result: &mut [u8]) -> Result<(), Error> {
```
Comment (NikVolf) [inline src/ext.rs]: U256 on value?
should convert then to bigendian and pass

File: src/ext.rs [hunk 2]
```diff
@@ -66,12 +69,12 @@ mod external {
...
 }
 
-pub fn suicide(refund: &[u8; 20]) {
+pub fn suicide(refund: &Address) {
     unsafe { external::suicide(refund.as_ptr()); }
 }
 
-pub fn create(endowment: &[u8; 32], code: &[u8]) -> Result<[u8; 20], Error> {
-    let mut result = [0u8; 20];
+pub fn create(endowment: &[u8; 32], code: &[u8]) -> Result<Address, Error> {
```
Comment (NikVolf) [inline src/ext.rs]: same with U256

**Comment by NikVolf** [comment]
block_hash should also return H256

**Comment by NikVolf** [comment]
gas_limit and fifficulty should return U256

**Comment by NikVolf** [comment]
Commit messages should be little more meaningful or i'll squash them

File: src/lib.rs [hunk 3]
```diff
@@ -133,28 +135,28 @@ impl<'a> ParamsView<'a> {
...
-		origin
+		Address::from(origin)
 	}
 
-	pub fn value(&self) -> [u8; 32] {
+	pub fn value(&self) -> U256 {
 		let mut value = [0u8; 32];
 		value.copy_from_slice(&self.raw[60..92]);
-		value
+		U256::from(value)
```
Comment (NikVolf) [inline src/lib.rs]: better explicitly use `from_big_endian` here

**Comment by fckt** [comment]
> Commit messages should be little more meaningful or i'll squash them
I know, I'm planning to squash them before merge

**Pull request merged**